### ISSUE TITLE
Unable to start thirdweb engine container

### ISSUE THREAD

**Issue by saez0pub** [thirdweb-dev/engine#502]
Last image update has a dependency problem.

stderr of thirdweb/engine:v0.0.36 startup:
```
npm WARN exec The following package was not found and will be installed: tsx@4.7.2
┌─────────────────────────────────────────────────────────┐
│  Update available 5.2.0 -> 5.12.1                       │
│  Run the following to update                            │
│    yarn add --dev prisma@latest                         │
│    yarn add @prisma/client@latest                       │
└─────────────────────────────────────────────────────────┘
npm notice 
npm notice New minor version of npm available! 10.2.4 -> 10.5.2
npm notice Changelog: <https://github.com/npm/cli/releases/tag/v10.5.2>
npm notice Run `npm install -g npm@10.5.2` to update!
npm notice 
(node:460) ExperimentalWarning: The Node.js specifier resolution flag is experimental. It could change or be removed at any time.
(Use `node --trace-warnings ...` to show where the warning was created)
(node:460) ExperimentalWarning: The Node.js specifier resolution flag is experimental. It could change or be removed at any time.
(Use `node --trace-warnings ...` to show where the warning was created)
file:///app/dist/utils/crypto.js:7
    return CryptoJS.AES.decrypt(data, password).toString(CryptoJS.enc.Utf8);
    ^

ReferenceError: CryptoJS is not defined
    at decrypt (file:///app/dist/utils/crypto.js:7:5)
    at withWalletConfig (file:///app/dist/db/configuration/getConfiguration.js:49:31)
    at getConfiguration (file:///app/dist/db/configuration/getConfiguration.js:147:12)
    at async getConfig (file:///app/dist/utils/cache/getConfig.js:13:24)
    at async queuedTxListener (file:///app/dist/worker/listeners/queuedTxListener.js:13:20)
    at async initWorker (file:///app/dist/worker/index.js:10:5)

Node.js v18.19.1
error Command failed with signal "SIGSEGV".
error Command failed with exit code 1.
```

seem related to https://github.com/thirdweb-dev/engine/commit/f2f2e40817571c67de1063d99afcd1bb9f037cfa

**Comment by farhanW3** [comment]
hi @saez0pub 

thanks for reporting this.

I am running the image (`v0.0.36`) locally and it seems to be working good.

```
postgres-1  | 
postgres-1  | PostgreSQL Database directory appears to contain a database; Skipping initialization
postgres-1  | 
postgres-1  | 2024-04-17 17:49:03.795 UTC [1] LOG:  starting PostgreSQL 16.1 (Debian 16.1-1.pgdg120+1) on aarch64-unknown-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit
postgres-1  | 2024-04-17 17:49:03.796 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
postgres-1  | 2024-04-17 17:49:03.796 UTC [1] LOG:  listening on IPv6 address "::", port 5432
postgres-1  | 2024-04-17 17:49:03.798 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
postgres-1  | 2024-04-17 17:49:03.800 UTC [29] LOG:  database system was shut down at 2024-02-28 00:51:50 UTC
postgres-1  | 2024-04-17 17:49:03.807 UTC [1] LOG:  database system is ready to accept connections
thirdweb-1  | yarn run v1.22.19
thirdweb-1  | $ yarn prisma:setup:prod && yarn start:run
thirdweb-1  | $ npx tsx ./dist/scripts/setup-db.js
thirdweb-1  | npm WARN exec The following package was not found and will be installed: tsx@4.7.2
thirdweb-1  | prisma:info Starting a postgresql pool with 10 connections.
thirdweb-1  | $ /app/node_modules/.bin/prisma migrate deploy --schema ./dist/prisma/schema.prisma
thirdweb-1  | Prisma schema loaded from dist/prisma/schema.prisma
thirdweb-1  | Datasource "db": PostgreSQL database "tt2", schema "public" at "postgres:5432"
postgres-1  | 2024-04-17 17:49:14.693 UTC [34] LOG:  could not receive data from client: Connection reset by peer
thirdweb-1  | 
thirdweb-1  | 31 migrations found in prisma/migrations
thirdweb-1  | 
thirdweb-1  | Applying migration `20240229013311_add_contract_event_subscriptions`
thirdweb-1  | Applying migration `20240313203128_add_transaction_receipts`
thirdweb-1  | Applying migration `20240323005129_add_idempotency_key_backward_compatible`
thirdweb-1  | Applying migration `20240327202800_add_keypairs`
thirdweb-1  | Applying migration `20240410015450_add_keypair_algorithm`
thirdweb-1  | Applying migration `20240411235927_add_keypair_label`
thirdweb-1  | 
thirdweb-1  | The following migrations have been applied:
thirdweb-1  | 
thirdweb-1  | migrations/
thirdweb-1  |   └─ 20240229013311_add_contract_event_subscriptions/
thirdweb-1  |     └─ migration.sql
thirdweb-1  |   └─ 20240313203128_add_transaction_receipts/
thirdweb-1  |     └─ migration.sql
thirdweb-1  |   └─ 20240323005129_add_idempotency_key_backward_compatible/
thirdweb-1  |     └─ migration.sql
thirdweb-1  |   └─ 20240327202800_add_keypairs/
thirdweb-1  |     └─ migration.sql
thirdweb-1  |   └─ 20240410015450_add_keypair_algorithm/
thirdweb-1  |     └─ migration.sql
thirdweb-1  |   └─ 20240411235927_add_keypair_label/
thirdweb-1  |     └─ migration.sql
thirdweb-1  |       
thirdweb-1  | All migrations have been successfully applied.
thirdweb-1  | $ /app/node_modules/.bin/prisma generate --schema ./dist/prisma/schema.prisma
thirdweb-1  | Prisma schema loaded from dist/prisma/schema.prisma
thirdweb-1  | 
thirdweb-1  | ✔ Generated Prisma Client (v5.2.0) to ./node_modules/@prisma/client in 293ms
thirdweb-1  | Start using Prisma Client in Node.js (See: https://pris.ly/d/client)
thirdweb-1  | ```
thirdweb-1  | import { PrismaClient } from '@prisma/client'
thirdweb-1  | const prisma = new PrismaClient()
thirdweb-1  | ```
thirdweb-1  | or start using Prisma Client at the edge (See: https://pris.ly/d/accelerate)
thirdweb-1  | ```
thirdweb-1  | import { PrismaClient } from '@prisma/client/edge'
thirdweb-1  | const prisma = new PrismaClient()
thirdweb-1  | ```
thirdweb-1  | 
thirdweb-1  | See other ways of importing Prisma Client: http://pris.ly/d/importing-client
thirdweb-1  | 
thirdweb-1  | ┌─────────────────────────────────────────────────────────┐
thirdweb-1  | │  Update available 5.2.0 -> 5.12.1                       │
thirdweb-1  | │  Run the following to update                            │
thirdweb-1  | │    yarn add --dev prisma@latest                         │
thirdweb-1  | │    yarn add @prisma/client@latest                       │
thirdweb-1  | └─────────────────────────────────────────────────────────┘
thirdweb-1  | npm notice 
thirdweb-1  | npm notice New minor version of npm available! 10.2.4 -> 10.5.2
thirdweb-1  | npm notice Changelog: <https://github.com/npm/cli/releases/tag/v10.5.2>
thirdweb-1  | npm notice Run `npm install -g npm@10.5.2` to update!
thirdweb-1  | npm notice 
thirdweb-1  | $ node --experimental-specifier-resolution=node ./dist/index.js
thirdweb-1  | (node:260) ExperimentalWarning: The Node.js specifier resolution flag is experimental. It could change or be removed at any time.
thirdweb-1  | (Use `node --trace-warnings ...` to show where the warning was created)
thirdweb-1  | (node:260) ExperimentalWarning: The Node.js specifier resolution flag is experimental. It could change or be removed at any time.
thirdweb-1  | (Use `node --trace-warnings ...` to show where the warning was created)
thirdweb-1  | prisma:info Starting a postgresql pool with 10 connections.
thirdweb-1  | [17:49:20 UTC] INFO: [Worker] Listening for queued transactions
thirdweb-1  | [17:49:20 UTC] INFO: [Worker] Listening for new configuration data
thirdweb-1  | [17:49:21 UTC] INFO: [Server] Engine server is listening on port 3005. Add to your dashboard: https://thirdweb.com/dashboard/engine?importUrl=http%3A%2F%2Flocalhost%3A3005.
thirdweb-1  | [17:49:21 UTC] INFO: [Server] Listening for updated transactions
thirdweb-1  | [17:49:21 UTC] INFO: [Worker] Listening for updated configuration data
thirdweb-1  | [17:49:21 UTC] INFO: [Worker] Listening for new webhooks data
thirdweb-1  | [17:49:21 UTC] INFO: [Worker] Listening for updated webhooks data
thirdweb-1  | [17:49:21 UTC] INFO: [Worker] Listening for indexed contracts
```

This my docker-compose.yml:

```
services:
  postgres:
    image: postgres:latest
    # restart: always
    volumes:
      - db_data:/var/lib/postgresql/data
    env_file:
      - ./.env
    ports:
      - "5432:5432" # Expose port 5432
    networks:
      - default
    deploy:
      resources:
        limits:
          cpus: "2"
          memory: 2G
      restart_policy:
        condition: on-failure

  thirdweb:
    image: thirdweb/engine:v0.0.36
    # restart: always
    env_file:
      - ./.env
    ports:
      - "3005:3005"
    depends_on:
      - postgres
    deploy:
      resources:
        limits:
          cpus: "1"
          memory: 1G
      restart_policy:
        condition: on-failure
volumes:
  db_data:
```

My Env File:

```
# DOCKER POSTGRES REQUIRED
POSTGRES_DATABASE_NAME=postgres
POSTGRES_USER=postgres
POSTGRES_PASSWORD=postgres

# THIRDWEB ENGINE REQUIRED
THIRDWEB_API_SECRET_KEY=<THIRDWEB_API_SECRET_KEY>
ENCRYPTION_PASSWORD=<ENCRYPTION_PASSWORD>
POSTGRES_CONNECTION_URL=postgres://postgres:postgres@postgres:5432/tt2?sslmode=disable&connection_limit=10
ADMIN_WALLET_ADDRESS=<ADMIN_WALLET_ADDRESS>
```

If you can share more details on your setup.

**Comment by saez0pub** [comment]
Hello,

I'm using a gcp kms for encryption.
And src/db/configuration/updateConfiguration.ts is calling encrypt from utils/crypto
CryptoJS is not present in yarn lock file.
CryptoJS is not imported in src/utils/crypto.ts
v0.0.35 is runing fine, but v0.0.36
Maybe this is the source of my problem.

**Comment by farhanW3** [comment]
Got it. I will dig in more. Thanks for the details.

**Comment by farhanW3** [comment]
Thank for reporting this @saez0pub 

We have released a new version `v0.0.37` which includes the fix. Let me know.

**Comment by saez0pub** [comment]
Thank you, I upgraded successfully.
